Automattic/newspack-popups
# Retrieve email, donation data from Mailchimp

### All Submissions:






### Changes proposed in this Pull Request:



#### Questions/issues:

1. This is using the Mailchimp API key provided for Newspack Newsletters plugin (`newspack_newsletters_mailchimp_api_key` option) – while concise, this introduces coupling between the two. Other way would be to add a similar option in this plugin's settings. WDYT?
2. Is donation data stored in Mailchimp as orders, and thus retrievable as the subscriber's total revenue? Assuming yes and waiting for more data.

Closes #304 

### How to test the changes in this Pull Request:

1. Create four inline campaigns and four matching segments: donors (D) and not donors (ND), subscribers (S) and not subscribers (NS)
2. Observe that by default only the two NS, ND campaigns are displayed in a new session
3. Create a newsletter (using Newspack Newsletters™ & Mailchimp) with a link to a post on the site
4. Visit that link in a new session, observe that only S, ND campaigns are visible
5. Now I have not found a simple way to create a dummy Mailchimp order, so I've tested that part by hardcoding a revenue value [in `api/segmentation/class-segmentation-client-data.php`](https://github.com/Automattic/newspack-popups/blob/c5460527612c460928f7fd600368e42887822c4d/api/segmentation/class-segmentation-client-data.php#L78).
6. Now that the test reader has some donation data saved, only D, S campaigns should be visible – on subsequent visits, since the MC data fetching is done asynchronously

_Note: you might need this filter to bypass SSL check on a local environment:_
```
add_filter(
	'http_request_args',
	function ( $r ) {
		$r['sslverify'] = false;
		return $r;
	}
);
```

### Other information:
